File: 001/Problem1.rs
use std::time::SystemTime;

fn main() {
    let start = SystemTime::now();
    problem();
    match start.elapsed() {
        Ok(elapsed) => {
            println!("{}", elapsed.as_nanos())
        }
        Err(e) => {
            println!("Error: {:?}", e)
        }
    }
}

fn problem() {
    let mut sum = 0;
    for n in 1..1000 {
        if n % 3 == 0 || n % 5 == 0 {
            sum += n;
        }
    }
    println!("{}", sum);
}

File: 001/Problem1.r
problem <- function() {
    sum <- 0
    for (i in 1:1000) {
        if (i %% 3 == 0 | i %% 5 == 0) {
            sum <- sum + i
        }
    }
    print(sum)
}
num_loops <- 1000
start.time <- Sys.time()
for(i in 1:num_loops) {
    problem()
}
end.time <- Sys.time()
time.taken <- end.time - start.time
print(time.taken / num_loops)